general/development/policies.md

```diff
@@ -0,0 +1,211 @@
+
+In general code should be written to avoid displaying interfaces which are removed, or adding new interfaces as the page loads.
+
+All Javascript must be accessible.
+
+:::info
+
+For more about this, see the [Javascript guide](/docs/guides/javascript).
+
+:::
+
+## Internationalisation
+
+Moodle works in over 84 languages because we pay great attention to keeping the language strings and locale information separate from the code, in language packs.
+
+The default language for all code, comments and documentation, however, is English (AU).
+
+:::info
+
+For more about this, see [String API](https://docs.moodle.org/dev/String_API)
+
+:::
+
+## Accessibility
+
+Moodle should work well for the widest possible range of people.
+
+:::info
+
+For more about this, see [Moodle Accessibility](/general/development/policies/accessibility).
+
+:::
+
+## Component library
+
+The Component library is a developer tool provided to help identify frequently-used user interface components, and encourage their re-use.
+
+It includes components from both Twitter Bootstrap, and from Moodle, and it displays these features using your current Moodle theme.
+
+:::info
+
+For more about this, see [Component library](https://docs.moodle.org/dev/Component_library).
+
+:::
+
+## Performance
+
+The load any Moodle site can cope with will, of course, depend on the server and network hardware that it is running on. However there are some features (intended especially for developers) that are discouraged on production sites for performance reasons.
+
+The most important property is scalability, so a small increase in the number of users, courses, activities in a course, and so on, only causes a correspondingly small increase in server load.
+
+:::info
+
+For more information and advice, see [Performance and scalability](https://docs.moodle.org/dev/Performance_and_scalability).
+
+:::
+
+## Database
+
+Moodle has a powerful database abstraction layer that we wrote ourselves, called [XMLDB](https://docs.moodle.org/dev/XMLDB_Documentation).  This lets the same Moodle code work on MySQL/MariaDB, PostgreSQL, MS SQL Server and Oracle. There are know issues when using Oracle, it is not fully supported and is not recommended for production sites.
+
+We have tools and API for [defining and modifying tables](https://docs.moodle.org/dev/DDL_functions) as well as [methods for getting data in and out](https://docs.moodle.org/dev/Data_manipulation_API) of the database.
+
+:::info
+
+For more about this, see [Moodle Database](https://docs.moodle.org/dev/Database) guidelines.
+
+:::
+
+## Events
+
+Moodle allows inter-module communication via **events**.  Modules can **trigger** specific events and other modules can choose to **handle/observe** those events.
```

Comment:
Err... this is not true.https://deploy-preview-157--peppy-zabaione-65df4a.netlify.app/general/development/policies/component-communication#event-observers

> The normal rules for inter-component communication apply.

Reply:
You're right! :-)  I've reviewed it :-)